Source/WebCore/platform/mediastream/gstreamer/GStreamerMediaStreamSource.cpp

```diff
@@ -370,7 +383,7 @@ class InternalSource final : public MediaStreamTrackPrivate::Observer,
             memset(data.data(), 0, yOffset);
             memset(data.data() + yOffset, 128, data.size() - yOffset);
         }
-        gst_buffer_add_video_meta_full(buffer.get(), GST_VIDEO_FRAME_FLAG_NONE, GST_VIDEO_FORMAT_I420, m_lastKnownSize.width(), m_lastKnownSize.height(), 3, info.offset, info.stride);
+        gst_buffer_add_video_meta_full(buffer.get(), GST_VIDEO_FRAME_FLAG_NONE, GST_VIDEO_FORMAT_I420, width, height, 3, info.offset, info.stride);
```

Comment:
if this a software filled buffer, perhaps there's no need of video meta.

Reply:
I dunno, we have offsets/stride info, I see no reason not to include it in a meta. Or would that have a perf impact?

Reply:
perf impact, minimal at most, I  just  wondered about removing non-functional code. But it doesn't harm either.

Reply:
Ok, I think we can revisit this part later on. It's a bit of a corner-case anyway because this code is about disabled video track handling. Most of the time users want an enabled track :)